qutebrowser/browser/inspector.py

```diff
@@ -52,18 +54,50 @@ class WebInspectorError(Exception):
 
 class AbstractWebInspector(QWidget):
 
-    """A customized WebInspector which stores its geometry."""
+    """A customized WebInspector which stores its geometry.
 
-    def __init__(self, parent=None):
+    Attributes:
+        _position: position of the inspector (right/left/top/bottom/window)
+        _splitter: InspectorSplitter where the inspector can be placed.
+
+    Signals:
+        closed: Emitted when the inspector is closed.
+    """
+
+    closed = pyqtSignal()
+    _position = None
+    _splitter = None
+
+    def __init__(self, splitter, parent=None):
         super().__init__(parent)
         self._widget = None
         self._layout = miscwidgets.WrapperLayout(self)
-        self._load_state_geometry()
+        self._splitter = splitter
 
     def _set_widget(self, widget):
         self._widget = widget
         self._layout.wrap(self, widget)
 
+    def set_position(self, position):
+        """Set the position of the inspector."""
+        if position != self._position:
+            if self._position == 'window':
+                self._save_state_geometry()
+
+            self._position = position
+
+            if position is None:
+                self.hide()
+                self.deleteLater()
+                self.closed.emit()
+            elif position == 'window':
+                self.hide()
+                self.setParent(None)
+                self._load_state_geometry()
+            else:
+                self._splitter.set_inspector(self, position)
+            self.show()
```

Comment:
Why is this called when position is None?

Reply:
the inspector has to do some cleanup, especially if it is inside a window which it needs to close.

Reply:
By "this" I meant the `self.show()` line. Surely it does not do clean-up in a function called `show`?

Reply:
Oh, that :)
Yes, I think you're right, it's not needed in case of None.